File: backend/ProyectHub.Models/Project.cs
namespace ProyectHub.Models
{
    public class Project
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Author { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public string Stack { get; set; } = string.Empty;
        public string Tags { get; set; } = string.Empty;
        public string Images { get; set; } = string.Empty;
        public string ImageUrl { get; set; } = string.Empty;
        public string DemoUrl { get; set; } = string.Empty;
        public string RepoUrl { get; set; } = string.Empty;
        public string CreatedAt { get; set; } = string.Empty;
    }
}

File: frontend/src/types/project.ts
export type GitCommit = {
  hash: string;
  message: string;
  author: string;
  date: string;
  filesChanged: number;
};

export type FileItem = {
  name: string;
  type: 'file' | 'folder';
  path: string;
  size?: number;
  lastModified?: string;
  children?: FileItem[];
};

export type FileContent = {
  path: string;
  content: string;
  language: string;
  size: number;
  lastModified: string;
};

export type GitRepository = {
  lastCommit?: GitCommit;
  structure?: FileItem[];
  totalCommits?: number;
  contributors?: number;
};

export type Project = {
  id: string; // Cambiado a string para compatibilidad con Supabase UUID
  title: string;
  description: string;
  repoUrl?: string;
  demoUrl?: string;
  imageUrl?: string;
  images?: string[]; // Galería de imágenes
  category?: string;
  createdAt?: string;
  stack?: string[];
  author?: string; // Autor del proyecto
  tags?: string[]; // Tags adicionales
  repository?: GitRepository;
  isDeleted?: boolean;
};